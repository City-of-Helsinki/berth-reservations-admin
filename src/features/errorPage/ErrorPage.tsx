import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './errorPage.module.scss';
import Text from '../../common/text/Text';
import Icon from '../../common/icons/Icon';
import ExternalLink from '../../common/externalLink/ExternalLink';
import Layout from '../../common/layout/Layout';
import Header from '../../common/header/Header';
import HelsinkiLogo from '../../common/helsinkiLogo/HelsinkiLogo';

const ErrorPage = () => {
  const { t } = useTranslation();

  return (
    <Layout
      header={
        <Header>
          <HelsinkiLogo size="large" color="white" />
        </Header>
      }
    >
      <div className={styles.container}>
        <div className={styles.innerContainer}>
          <Icon shape="IconExclamationCircle" size="large" />
          <Text as="h1" className={styles.heading}>
            {t('errorPage.heading')}
          </Text>
          <Text className={styles.text}>{t('errorPage.text')}</Text>
          <ExternalLink href="/" target="_self">
            {t('errorPage.linkText')}
          </ExternalLink>
        </div>
      </div>
    </Layout>
  );
};

export default ErrorPage;
