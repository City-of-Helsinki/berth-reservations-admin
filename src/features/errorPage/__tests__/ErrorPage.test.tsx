import React from 'react';
import { shallow } from 'enzyme';
import { HashRouter } from 'react-router-dom';

import ErrorPage from '../ErrorPage';

describe('ErrorPage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <HashRouter>
        <ErrorPage />
      </HashRouter>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
