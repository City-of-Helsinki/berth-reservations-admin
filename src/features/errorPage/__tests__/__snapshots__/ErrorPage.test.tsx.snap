// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly 1`] = `"<main class=\\"layout noSidebar\\"><header class=\\"header\\"><header class=\\"header\\"><svg class=\\"helsinkiLogo large white\\">helsinki-logo.svg</svg></header></header><div class=\\"content\\"><div class=\\"container\\"><div class=\\"innerContainer\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path d=\\"M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 12v2h-2v-2h2zm0-10v8h-2V6h2z\\" fill=\\"currentColor\\"></path></g></svg><h1 class=\\"text standard h1 heading\\">Virhetilanne</h1><span class=\\"text standard span text\\">Pahoittelumme, järjestelmässä ilmeni virhe. Voit yrittää uudelleen, mutta jos sama toistuu, kerro siitä meille niin korjaamme asian.</span><a class=\\"link default\\" href=\\"/\\" target=\\"_self\\" rel=\\"noopener noreferrer\\">Etusivulle</a></div></div></div></main>"`;
